Choma similarity_search_with_score (and similar methods) don't populate the document.id property of the returned documents.

### Checked other resources

- [X] I added a very descriptive title to this issue.
- [X] I searched the LangChain documentation with the integrated search.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I am sure that this is a bug in LangChain rather than my code.
- [X] The bug is not resolved by updating to the latest stable version of LangChain (or the specific integration package).

### Example Code

I want to get the document id from the documents I find with `similarity_search_with_score` but the id's aren't being set to the documents when they are created.

Here is my test code:

```python
def test_basic_operations_with_langchain():
    token = os.environ.get('CHROMA_TOKEN')
    client = chromadb.HttpClient(
        settings=Settings(chroma_client_auth_provider="chromadb.auth.token_authn.TokenAuthClientProvider",
                          chroma_client_auth_credentials=token))
    test_collection_name = "test_collection"

    try:
        client.delete_collection(test_collection_name)
    except:  # noqa: E722
        pass

    collection = client.get_or_create_collection(test_collection_name)
    assert collection.name == test_collection_name
    assert collection.count() == 0

    EMBEDDING_MODEL_NAME = "sentence-transformers/all-mpnet-base-v2"
    EMBEDDING_MODEL_CHUNK_SIZE = 384

    embedding_model = HuggingFaceEmbeddings(
        model_name=EMBEDDING_MODEL_NAME,
        # multi_process=True,   # I found this causes crashes and slowness
        model_kwargs={"device": "cuda" if torch.cuda.is_available() else "cpu"},
        encode_kwargs={
            "normalize_embeddings": True
        },  # set True for cosine similarity
    )
    vector_db = Chroma(
        embedding_function=embedding_model,
        client=client,
        collection_name=test_collection_name,
        collection_metadata={"hnsw:space": "cosine"},
    )
    # Make new uid
    id = str(uuid.uuid4())
    test_doc = Document("hello world")
    test_doc.id = id
    test_doc.metadata["test"] = "test"
    vector_db.add_documents([test_doc])
    assert collection.count() == 1

    docs = vector_db.similarity_search_with_score("hello world", 1)
    assert len(docs) == 1
    assert docs[0][0].page_content == "hello world"
    assert docs[0][0].id == id

    client.delete_collection(test_collection_name)
```


### Error Message and Stack Trace (if applicable)


Test output:
```
=================================== FAILURES ===================================
_____________________ test_basic_operations_with_langchain _____________________

    def test_basic_operations_with_langchain():
        token = os.environ.get('CHROMA_TOKEN')
        client = chromadb.HttpClient(
            settings=Settings(chroma_client_auth_provider="chromadb.auth.token_authn.TokenAuthClientProvider",
                              chroma_client_auth_credentials=token))
        test_collection_name = "test_collection"
    
        try:
            client.delete_collection(test_collection_name)
        except:  # noqa: E722
            pass
    
        collection = client.get_or_create_collection(test_collection_name)
        assert collection.name == test_collection_name
        assert collection.count() == 0
    
        EMBEDDING_MODEL_NAME = "sentence-transformers/all-mpnet-base-v2"
        EMBEDDING_MODEL_CHUNK_SIZE = 384
    
        embedding_model = HuggingFaceEmbeddings(
            model_name=EMBEDDING_MODEL_NAME,
            # multi_process=True,   # I found this causes crashes and slowness
            model_kwargs={"device": "cuda" if torch.cuda.is_available() else "cpu"},
            encode_kwargs={
                "normalize_embeddings": True
            },  # set True for cosine similarity
        )
        vector_db = Chroma(
            embedding_function=embedding_model,
            client=client,
            collection_name=test_collection_name,
            collection_metadata={"hnsw:space": "cosine"},
        )
        # Make new uid
        id = str(uuid.uuid4())
        test_doc = Document("hello world")
        test_doc.id = id
        test_doc.metadata["test"] = "test"
        vector_db.add_documents([test_doc])
        assert collection.count() == 1
    
        docs = vector_db.similarity_search_with_score("hello world", 1)
        assert len(docs) == 1
        assert docs[0][0].page_content == "hello world"
>       assert docs[0][0].id == id
E       AssertionError: assert None == '808971a7-f042-4306-864b-97af29c4837d'
E        +  where None = Document(metadata={'test': 'test'}, page_content='hello world').id

code/tests/test_basic_chroma_setup.py:99: AssertionError
=============================== warnings summary ===============================
<frozen importlib._bootstrap>:488
  <frozen importlib._bootstrap>:488: DeprecationWarning: Type google._upb._message.MessageMapContainer uses PyType_Spec with a metaclass that has custom tp_new. This is deprecated and will no longer be allowed in Python 3.14.

<frozen importlib._bootstrap>:488
  <frozen importlib._bootstrap>:488: DeprecationWarning: Type google._upb._message.ScalarMapContainer uses PyType_Spec with a metaclass that has custom tp_new. This is deprecated and will no longer be allowed in Python 3.14.

code/tests/test_basic_chroma_setup.py::test_basic_operations_with_langchain
  /projects/myproj/.venv/lib/python3.12/site-packages/transformers/tokenization_utils_base.py:1601: FutureWarning: `clean_up_tokenization_spaces` was not set. It will be set to `True` by default. This behavior will be depracted in transformers v4.45, and will be then set to `False` by default. For more details check this issue: https://github.com/huggingface/transformers/issues/31884
    warnings.warn(

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
FAILED code/tests/test_basic_chroma_setup.py::test_basic_operations_with_langchain
======================== 1 failed, 3 warnings in 6.21s =========================
Finished running tests!

```
I would expect the result:

```
=============================== warnings summary ===============================
<frozen importlib._bootstrap>:488
  <frozen importlib._bootstrap>:488: DeprecationWarning: Type google._upb._message.MessageMapContainer uses PyType_Spec with a metaclass that has custom tp_new. This is deprecated and will no longer be allowed in Python 3.14.

<frozen importlib._bootstrap>:488
  <frozen importlib._bootstrap>:488: DeprecationWarning: Type google._upb._message.ScalarMapContainer uses PyType_Spec with a metaclass that has custom tp_new. This is deprecated and will no longer be allowed in Python 3.14.

code/tests/test_basic_chroma_setup.py::test_basic_operations_with_langchain
  /projects/myproj/.venv/lib/python3.12/site-packages/transformers/tokenization_utils_base.py:1601: FutureWarning: `clean_up_tokenization_spaces` was not set. It will be set to `True` by default. This behavior will be depracted in transformers v4.45, and will be then set to `False` by default. For more details check this issue: https://github.com/huggingface/transformers/issues/31884
    warnings.warn(

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
======================== 1 passed, 3 warnings in 7.26s =========================
Finished running tests!
```


### Description

I am using:

```
langchain==0.2.15
langchain-chroma==0.1.3
langchain-community==0.2.15
langchain-core==0.2.37
langchain-huggingface==0.0.3
langchain-text-splitters==0.2.2
chromadb==0.5.3
```

I am calling `vector_db.similarity_search_with_score(...)` or `vector_db.similarity_search(...)` and the documents returned never have an id.

I can't see a way of getting the documents give me an id. I though about adding a kwargs include ids. But the documentation for the underlying query says it always returns ids.  I think the issue is in your method:

```
def _results_to_docs_and_scores(results: Any) -> List[Tuple[Document, float]]:
    return [
        # TODO: Chroma can do batch querying,
        # we shouldn't hard code to the 1st result
        (Document(page_content=result[0], metadata=result[1] or {}), result[2])
        for result in zip(
            results["documents"][0],
            results["metadatas"][0],
            results["distances"][0],
        )
    ]
```
from: https://github.com/langchain-ai/langchain/blob/51c4393298801b75f98cf8a060376237a734eac1/libs/partners/chroma/langchain_chroma/vectorstores.py#L43-L53

It doesn't use the id.

This fix should be very easy:

```
def _results_to_docs_and_scores(results: Any) -> List[Tuple[Document, float]]:
    return [
        # TODO: Chroma can do batch querying,
        # we shouldn't hard code to the 1st result
        (Document(id =result[0], page_content=result[1], metadata=result[2] or {}), result[3])
        for result in zip(
            results["ids"][0],
            results["documents"][0],
            results["metadatas"][0],
            results["distances"][0],
        )
    ]

```

### System Info

System Information
------------------
> OS:  Linux
> OS Version:  #1 SMP Fri Mar 29 23:14:13 UTC 2024
> Python Version:  3.12.4 (main, Jul 19 2024, 17:20:16) [GCC 11.4.1 20231218 (Red Hat 11.4.1-3)]

Package Information
-------------------
> langchain_core: 0.2.37
> langchain: 0.2.15
> langchain_community: 0.2.15
> langsmith: 0.1.108
> langchain_chroma: 0.1.3
> langchain_huggingface: 0.0.3
> langchain_text_splitters: 0.2.2

Optional packages not installed
-------------------------------
> langgraph
> langserve

Other Dependencies
------------------
> aiohttp: 3.10.5
> async-timeout: Installed. No version info available.
> chromadb: 0.5.3
> dataclasses-json: 0.6.7
> fastapi: 0.112.2
> httpx: 0.27.2
> huggingface-hub: 0.24.6
> jsonpatch: 1.33
> numpy: 1.26.4
> orjson: 3.10.7
> packaging: 24.1
> pydantic: 2.8.2
> PyYAML: 6.0.2
> requests: 2.32.3
> sentence-transformers: 3.0.1
> SQLAlchemy: 2.0.32
> tenacity: 8.5.0
> tokenizers: 0.19.1
> transformers: 4.44.2
> typing-extensions: 4.12.2

--- a/libs/partners/couchbase/langchain_couchbase/vectorstores.py
+++ b/libs/partners/couchbase/langchain_couchbase/vectorstores.py
@@ -559,12 +559,13 @@ class CouchbaseVectorStore(VectorStore):
             # Parse the results
             for row in search_iter.rows():
                 text = row.fields.pop(self._text_key, "")
+                id = row.id
 
                 # Format the metadata from Couchbase
                 metadata = self._format_metadata(row.fields)
 
                 score = row.score
-                doc = Document(page_content=text, metadata=metadata)
+                doc = Document(id=id, page_content=text, metadata=metadata)
                 docs_with_score.append((doc, score))
 
         except Exception as e:

--- a/libs/partners/couchbase/pyproject.toml
+++ b/libs/partners/couchbase/pyproject.toml
@@ -4,7 +4,7 @@ build-backend = "poetry.core.masonry.api"
 
 [tool.poetry]
 name = "langchain-couchbase"
-version = "0.2.0"
+version = "0.2.1"
 description = "An integration package connecting Couchbase and LangChain"
 authors = []
 readme = "README.md"

--- a/libs/partners/couchbase/tests/integration_tests/test_vector_store.py
+++ b/libs/partners/couchbase/tests/integration_tests/test_vector_store.py
@@ -193,6 +193,7 @@ class TestCouchbaseVectorStore:
         time.sleep(SLEEP_DURATION)
 
         output = vectorstore.similarity_search("foo", k=1)
+        assert output[0].id == "a"
         assert output[0].page_content == "foo"
         assert output[0].metadata["a"] == 1
 
@@ -364,3 +365,32 @@ class TestCouchbaseVectorStore:
 
         assert result == hybrid_result
         assert score <= hybrid_score
+
+    def test_id_in_results(self, cluster: Any) -> None:
+        """Test that the id is returned in the result documents."""
+
+        texts = [
+            "foo",
+            "bar",
+            "baz",
+        ]
+
+        metadatas = [{"a": 1}, {"b": 2}, {"c": 3}]
+
+        vectorstore = CouchbaseVectorStore(
+            cluster=cluster,
+            embedding=ConsistentFakeEmbeddings(),
+            index_name=INDEX_NAME,
+            bucket_name=BUCKET_NAME,
+            scope_name=SCOPE_NAME,
+            collection_name=COLLECTION_NAME,
+        )
+
+        ids = vectorstore.add_texts(texts, metadatas=metadatas)
+        assert len(ids) == len(texts)
+
+        # Wait for the documents to be indexed
+        time.sleep(SLEEP_DURATION)
+
+        output = vectorstore.similarity_search("foo", k=1)
+        assert output[0].id == ids[0]

